### ISSUE TITLE
Not detected return inside subshell

### ISSUE THREAD

**Issue by mikhailnov** [koalaman/shellcheck#2729]
"return 0" in the followin code is misused:

```bash
#!/bin/bash
# returns 0 if at least one user exists
_has_local_users(){
	if [ -z "$(ls -A /home)" ]; then
		return 1
	fi
	find /home -maxdepth 1 -type d | grep -v '^/home$' |
	while read -r line
	do
		local user
		user="$(LC_ALL=C stat --printf='%U' "$line")"
		# lost+found on ext4 belongs to root
		# UNKNOWN means that a user with such UID does not exist
		if [ "$user" != UNKNOWN ] && [ "$user" != root ]; then
			return 0
		fi
	done
	return 1
}
```

Its author (me) wanted to make a return from function, but made a return from subshell. that does not make sense.
shellcheck does not give any warnings.
